### DISCUSSION TITLE
Support for View Models

### DISCUSSION THREAD

**Discussion by vJoeyz** [statamic-rad-pack/runway#611] [Feature Requests]
Support for something like view models, perhaps in the same way [statamic](https://statamic.dev/view-models) does it. 

I've made a very very very rough poc [here](https://github.com/statamic-rad-pack/runway/compare/7.x...vJoeyz:runway:view-models-poc). I wasn't really sure where to put all that stuff. However, it does seem to work and hydrates the cascade given the view model defined in runway.php. I'm even less sure of how it can work with relations and caching.

**Comment by duncanmcclean** [comment]
If it works, feel free to submit a pull request and I can build on top of it when I get a chance.

**Comment by hlz** [comment]
Came here since I need support for View Models and thus had the same question. Any updates in case of View Models or should I try to get the POC of vJoeyz working in my project?

Or are there any other (good) methods to get some database data (aggregate data of a Runway resource) in the view layer?

**Reply by duncanmcclean** [reply]
No updates, sorry!

> Or are there any other (good) methods to get some database data (aggregate data of a Runway resource) in the view layer?

The best way I can think of is a custom tag.

**Reply by hlz** [reply]
No problem Duncan! A custom tag certainly can do the job.
